APMrover2/mode_guided.cpp

```diff
@@ -31,11 +31,10 @@ void ModeGuided::update()
                 calc_steering_to_waypoint(_reached_destination ? rover.current_loc : _origin, _destination, _reversed);
                 calc_throttle(calc_reduced_speed_for_turn_or_distance(_reversed ? -_desired_speed : _desired_speed), true, true);
             } else {
-                if (rover.is_boat() && !start_loiter()) {
-                    stop_vehicle();
-                } else {
-                    stop_vehicle();
+                if (rover.is_boat()) {
```

Comment:
https://github.com/ArduPilot/ardupilot/issues/10188
@rmackay9 san... I am changing the comment on this subject.

Reply:
Thank you. I think there is no need to call stop_vehicle() even once..But it's just my opinion. Better way should be adapted.